crates/relayer/src/chain/namada/tx.rs

```diff
@@ -67,17 +71,44 @@ impl NamadaChain {
             txs.push((tx, signing_data));
         }
         let (mut tx, signing_data) = tx::build_batch(txs).map_err(NamadaError::namada)?;
+        let signing_data = signing_data.first().expect("SigningData should exist");
+
+        // Estimate the fee with dry-run
+        let (fee_token, gas_limit, fee_amount) =
+            self.estimate_fee(tx.clone(), &tx_args, signing_data)?;
```

Comment:
Not all nodes should be required to enable dry running txs via their exposed RPC servers, since these may pose a DoS vector. `estimate_fee` should be expected to fail with a nonexistent RPC error, at which point we could fallback to another gas estimation mechanism (e.g. median values that have been benchmarked during a release).

Reply:
That makes sense. I'll fix it.